diesel/src/macros/mod.rs

```diff
@@ -45,6 +51,26 @@ macro_rules! column {
         {
         }
 
+        impl<Left, Right, FK> $crate::expression::SelectableExpression<
```

Comment:
> No, that trait doesn't exist and it would not be possible to write it today.

Using specialization this seems to be [possible](https://play.rust-lang.org/?gist=e993c9ae92e72235039cb848c3bbee48&version=nightly&backtrace=0).  Currently there seems to be a problem with the resolution of associated types when using specialization. See the following [minimal example](https://play.rust-lang.org/?gist=9161d4394c8278c70a287270752ab7a1&version=nightly&backtrace=0). Because of this the explicit specialization from line 27 to line 37 and from line 66 to line 72 is needed.